pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -432,56 +431,39 @@ func (m *ManagerImpl) GetCapacity() (v1.ResourceList, v1.ResourceList, []string)
 	return capacity, allocatable, deletedResources
 }
 
-// checkpointData struct is used to store pod to device allocation information
-// and registered device information in a checkpoint file.
-// TODO: add version control when we need to change checkpoint format.
-type checkpointData struct {
-	PodDeviceEntries  []podDevicesCheckpointEntry
-	RegisteredDevices map[string][]string
-}
-
 // Checkpoints device to container allocation information to disk.
 func (m *ManagerImpl) writeCheckpoint() error {
 	m.mutex.Lock()
-	data := checkpointData{
-		PodDeviceEntries:  m.podDevices.toCheckpointData(),
-		RegisteredDevices: make(map[string][]string),
-	}
+	data := NewDevicePluginCheckpoint()
+	data.(*checkpointData).podDeviceEntries = m.podDevices.toCheckpointData()
```

Comment:
can we pass podDevices and healthyDevices (or whatever we need) to NewDevicePluginCheckpoint rather than casting to checkpointData to reveal non-exported fields?

Reply:
sure.